### ISSUE TITLE
Attempt to replication object in case of policy with only one copy

### ISSUE THREAD

**Issue by anatoly-bogatyrev** [nspcc-dev/neofs-node#1184]
Version: v0.27.5
Node: karen - st01.testnet.fs.neo.org:8080

Log: 
```
фев 16 07:33:37 karen neofs-storage[735]: 2022-02-16T07:33:37.951Z        error        replicator/process.go:82        could not replicate object        {"component": "Object Replicator", "node": "0296416a8da7e2d144688d3f1f7af9321b88d12b0497c9b630d92a0f483610f8d5", "object": "5HmHKNKsPgyGnp37yCNVaVm7k9o3jvVeAbXXoJj6RHax/GHKEnSzYS2BRgYVvUWiDzRxsvVUYdmPWii27jEFaE5Fk", "error": "(*putsvc.RemoteSender) could not send object: (*putsvc.remoteTarget) could not put object to [/dns4/st04.testnet.fs.neo.org/tcp/8080]: status: code = 1024 message = (*response.ClientMessageStreamer) could not send the request: access to operation PUT is denied by basic ACL check"}
фев 16 07:33:37 karen neofs-storage[735]: 2022-02-16T07:33:37.951Z        error        replicator/process.go:82        could not replicate object        {"component": "Object Replicator", "node": "0296416a8da7e2d144688d3f1f7af9321b88d12b0497c9b630d92a0f483610f8d5", "object": "5HmHKNKsPgyGnp37yCNVaVm7k9o3jvVeAbXXoJj6RHax/FWvMsrjt8KA6fsxYLe9kVXutuRvp5AoDo7uMTAgR9359", "error": "(*putsvc.RemoteSender) could not send object: (*putsvc.remoteTarget) could not put object to [/dns4/st04.testnet.fs.neo.org/tcp/8080]: status: code = 1024 message = (*response.ClientMessageStreamer) could not send the request: access to operation PUT is denied by basic ACL check"}
фев 16 07:33:38 karen neofs-storage[735]: 2022-02-16T07:33:38.016Z        error        replicator/process.go:82        could not replicate object        {"component": "Object Replicator", "node": "0296416a8da7e2d144688d3f1f7af9321b88d12b0497c9b630d92a0f483610f8d5", "object": "5HmHKNKsPgyGnp37yCNVaVm7k9o3jvVeAbXXoJj6RHax/VsgDnWmpFww6XDP2A62YtKnf9ex7FDDFJshc6aX1G3F", "error": "(*putsvc.RemoteSender) could not send object: (*putsvc.remoteTarget) could not put object to [/dns4/st04.testnet.fs.neo.org/tcp/8080]: status: code = 1024 message = (*response.ClientMessageStreamer) could not send the request: access to operation PUT is denied by basic ACL check"}
```

Container:
```
container ID: 5HmHKNKsPgyGnp37yCNVaVm7k9o3jvVeAbXXoJj6RHax
version: 2.11
owner ID: NQgjuLbEzWYqzs1krT7YrDtSDdQU8JFdPS
basic ACL: 3078316338633863 (private)
attribute: Timestamp=1643257324 (2022-01-27 07:22:04 +0300 MSK)
nonce: ace62fa5-31d2-4f6b-af18-b7d3a9460213
placement policy:
REP 1 IN X
CBF 1
SELECT 1 FROM F AS X
FILTER UN-LOCODE EQ DE FKS AS F
```

Node location st01.testnet.fs.neo.org:8080: UN-LOCODE:DE FKS

**Comment by carpawell** [comment]
I rechecked placement for the current network map in the `testnet` and found out that `s04` is a preferred node for that object. However, it is impossible for `s04` to approve that object due to ACL check errors.
This is most likely ([this](https://github.com/nspcc-dev/neofs-node/issues/1246#issuecomment-1070536098) would bring more detailed info) due to the fact that `s04` could not prove that `s01` has ever been in the container: node is able to check container state only for the current and the previous epochs ([Issue solving](https://github.com/nspcc-dev/neofs-contract/issues/224) would help but not prevent such issues at all).

So basically: if the nodes somehow have been down for more than two epochs and container placement is not shared between them for now, migration processes for such containers are impossible.

Strictly speaking, described behavior is not a bug, it is a correct node reaction on another node's being down for too many epochs.

**Comment by alexvanin** [comment]
There are more objects that can't be replicated due to the same error
```
"object": "A12rhV2i6P6Fsc3U98EKztRhaqXhdfqy83e4nRGGwCjk/8rXEt5NXbyM4TjTvukbfd8apJ27VzvDK3yMz2qUEeaFd", "error": ...could not send the request: access to operation PUT is denied by basic ACL check"}
```

And this container has proper placement policy:
```
container ID: A12rhV2i6P6Fsc3U98EKztRhaqXhdfqy83e4nRGGwCjk
version: 2.10
owner ID: NceCu75mcC9pCu9D8ykj1jyZQGjzXcitGZ
basic ACL: 3078316662663863 (public-read)
attribute: Timestamp=1637843077 (2021-11-25 15:24:37 +0300 MSK)
nonce: 679bbeca-2266-40af-8074-07fa31622437
placement policy:
REP 2 IN X
CBF 1
SELECT 4 FROM F AS X
FILTER Deployed EQ NSPCC AS F
```

**Comment by alexvanin** [comment]
What we can do to decrease amount of such failures:
1. make a priority in object listing at policer component to replicate objects that are out of container first,
2. provide proof of placement in network map.

As for the second option, we might implement such scheme:
![image](https://user-images.githubusercontent.com/6323066/162148634-98e6b597-635a-4766-8659-63d4ca8b496f.png)

Policer can retry object put by attaching `proof` that it was a container node a while ago. This `proof` is a epoch number when node was a part of the container. This number can be transported as x-header. Old nods will not process it and behaviour stays the same. New nodes might check that proof and process request with `SYSTEM` role.

Request sending node should find proper network map copy and receiving node must also have the network map from this epoch. They can use local caches (which they lack of as for now). With neofs-contract v0.15.0 (https://github.com/nspcc-dev/neofs-contract/issues/224)  nodes may fetch up to 9 previous epoch. And with https://github.com/nspcc-dev/neo-go/issues/2407 it will be possible to make historic invocation and find any network map from the past.

**Comment by carpawell** [comment]
@alexvanin, we discussed a little some possible solutions to that problem with @cthulhu-rider and decided that almost all suggestions only help but do not solve the problem. It seems that only historic invocations (and maybe some netmap caching for preventing multiple RPC calls to find the required epoch) solve the issue completely but requires some significant changes from the API to the Morph client in the node.

Some questions:
1. How should we describe that logic in the API? Should *every* node implementation do that checks, implement caching mechanism, be able to retry `Put` with `proof`?
2. Do C# RPC nodes support historic invocations?
3. Do we have any information about historic invocation execution time on a `neo-go` RPC node? 

/cc @fyrchik

**Comment by roman-khimov** [comment]
The scheme outlined by @alexvanin is pretty easy to implement now, because historic invokers are the same API-wise as the current ones. But it'd be nice to have block numbers as well, otherwise we have a problem of epoch to block matching.

**Comment by roman-khimov** [comment]
This can be a bug of its own as well. Consider a malicious node that once was a part of netmap (or lost its keys). With this proof it could push almost any objects into container without even being a part of it.
